### ISSUE TITLE
stream_socket_client(): php_network_getaddresses: getaddrinfo failed: Name or service not known

### ISSUE THREAD

**Issue by nadavkav** [wiris/moodle-filter_wiris#17]
I am getting the following error on a quiz page that uses WIRIS with ATTO

uncaught exception: stream_socket_client(): php_network_getaddresses: getaddrinfo failed: Name or service not known (errno: 2) in /var/www/moodle/filter/wiris/integration/lib/sys/net/Socket.class.php at line #76

in file: /var/www/moodle/filter/wiris/integration/lib/com/wiris/plugin/impl/HttpImpl.class.php line 61
#0 /var/www/moodle/filter/wiris/integration/lib/php/Boot.class.php(769): com_wiris_plugin_impl_HttpImpl_0(Object(com_wiris_plugin_impl_HttpImpl), NULL, 'http://www.wiri...', 'stream_socket_c...')
#1 /var/www/moodle/filter/wiris/integration/lib/haxe/Http.class.php(532): _hx_lambda->execute('stream_socket_c...')
#2 /var/www/moodle/filter/wiris/integration/lib/php/Boot.class.php(769): haxe_Http_5(true, Object(com_wiris_plugin_impl_HttpImpl), Object(com_wiris_plugin_impl_HttpImpl), Array, Object(haxe_io_BytesOutput), true, 'stream_socket_c...')
#3 /var/www/moodle/filter/wiris/integration/lib/com/wiris/plugin/impl/HttpImpl.class.php(27): _hx_lambda->execute('stream_socket_c...')
#4 /var/www/moodle/filter/wiris/integration/lib/haxe/Http.class.php(444): com_wiris_plugin_impl_HttpImpl->onError('stream_socket_c...')
#5 /var/www/moodle/filter/wiris/integration/lib/haxe/Http.class.php(458): haxe_Http->customRequest(true, Object(haxe_io_BytesOutput), Object(sys_net_Socket), NULL)
#6 /var/www/moodle/filter/wiris/integration/lib/com/wiris/plugin/impl/HttpImpl.class.php(40): haxe_Http->request(true)
#7 /var/www/moodle/filter/wiris/integration/lib/com/wiris/plugin/impl/RenderImpl.class.php(252): com_wiris_plugin_impl_HttpImpl->request(true)
#8 /var/www/moodle/filter/wiris/integration/lib/com/wiris/plugin/impl/TextFilter.class.php(58): com_wiris_plugin_impl_RenderImpl->showImage('17b181be2d99c11...', 'filterMath(Object(com_wiris_plugin_impl_TextFilterTags), '

<...', Array, true)
#11 /var/www/moodle/filter/wiris/integration/lib/com/wiris/plugin/impl/TextServiceImpl.class.php(9): com_wiris_plugin_impl_TextFilter->filter('

<...', Array)
#12 /var/www/moodle/filter/wiris/filter.php(113): com_wiris_plugin_impl_TextServiceImpl->filter('

<...', Array)
#13 /var/www/moodle/lib/filterlib.php(176): filter_wiris->filter('

<...', Array)
#14 /var/www/moodle/lib/filterlib.php(218): filter_manager->apply_filter_chain('

<...', Array, Array, NULL)
#15 /var/www/moodle/lib/weblib.php(1282): filter_manager->filter_text('

<...', Object(context_module), Array)
#16 /var/www/moodle/question/type/questionbase.php(364): format_text('

<...', '1', Array)
#17 /var/www/moodle/question/type/questionbase.php(387): question_definition->format_text('

<...', '1', Object(question_attempt), 'question', 'generalfeedback', '201107')
#18 /var/www/moodle/question/type/rendererbase.php(177): question_definition->format_generalfeedback(Object(question_attempt))
#19 /var/www/moodle/question/type/rendererbase.php(120): qtype_renderer->general_feedback(Object(question_attempt))
#20 /var/www/moodle/question/engine/renderer.php(411): qtype_renderer->feedback(Object(question_attempt), Object(mod_quiz_display_options))
#21 /var/www/moodle/question/engine/renderer.php(113): core_question_renderer->outcome(Object(question_attempt), Object(qbehaviour_interactive_renderer), Object(qtype_numerical_renderer), Object(mod_quiz_display_options))
#22 /var/www/moodle/question/behaviour/behaviourbase.php(113): core_question_renderer->question(Object(question_attempt), Object(qbehaviour_interactive_renderer), Object(qtype_numerical_renderer), Object(mod_quiz_display_options), 2)
#23 /var/www/moodle/question/engine/questionattempt.php(829): question_behaviour->render(Object(mod_quiz_display_options), 2, Object(core_question_renderer), Object(qtype_numerical_renderer))
#24 /var/www/moodle/question/engine/questionusage.php(449): question_attempt->render(Object(mod_quiz_display_options), 2)
#25 /var/www/moodle/mod/quiz/attemptlib.php(1537): question_usage_by_activity->render_question('2', Object(mod_quiz_display_options), 2)
#26 /var/www/moodle/mod/quiz/attemptlib.php(1500): quiz_attempt->render_question_helper('2', true, Object(moodle_url), Object(theme_remui\output\mod_quiz_renderer), NULL)
#27 /var/www/moodle/mod/quiz/renderer.php(187): quiz_attempt->render_question('2', true, Object(theme_remui\output\mod_quiz_renderer), Object(moodle_url))
#28 /var/www/moodle/mod/quiz/renderer.php(56): mod_quiz_renderer->questions(Object(quiz_attempt), true, Array, 0, true, Object(mod_quiz_display_options))
#29 /var/www/moodle/mod/quiz/review.php(259): mod_quiz_renderer->review_page(Object(quiz_attempt), Array, 0, true, true, Object(mod_quiz_display_options), Array)
#30 {main}

**Comment by manuelwiris** [comment]
@nadavkav 

Do you experience this problem on any formula created by Atto?
Can you reproduce this problem during editing? 

Thanks,

**Comment by nadavkav** [comment]
It showed up only on a quiz review page only, and not on regular quiz and question editing pages.
We are behind FW and using Proxy to get out to the internet, and I thought you tried going directly to the internet, but we did not see any request blocked on our FW.
I tried to debug it but it was unclear to me what was causing this issue.

Eventually, I upgraded the filter/wiris and atto/plugin/wiris to latest versions and the issue was solved :-)
